### ISSUE TITLE
Persp-projectile activation instructions are unclear.

### ISSUE THREAD

**Issue by BrianHicks** [syl20bnr/spacemacs#495]
Under https://github.com/syl20bnr/spacemacs/tree/master/contrib/perspectives#persp-projectile, it says to "[put] (require 'persp-projectile) in your configuration files." I've tried this in `dotspacemacs/init`, `dotspacemacs/config`, and just before both functions. I've also tried including it in `dotspacemacs-configuration-layers`, and with `use-package` in `dotspacemacs/config`. Only the last one doesn't error, but it still doesn't open new perspectives when I switch projectile projects.

How can I fix this, and what else can I give you to help diagnose the problem?

**Comment by CestDiego** [comment]
Hi, I just looked up my config to see if I use `(require 'pers-projectile)` and I don't I must have removed it or sth after some upgrade made it invalid. The thing is that this is now lazy loaded when you use a custom perspective. So, for example `<kbd>SPC</kbd> P o e` opens a perspective with the init.el of spacemacs, afterwards any project you open with `<kbd>SPC</kbd> p s` (projectile-switch-project) will open a new project in a new perspective (I don't remember well why was this done this way. I guess that it's because the hability to create a new perspective after you open a project kinda overrides the projectile-switch-project function, so many people may not find that convenient.

In short, just add the contrib layer to the dotspacemacs-configuration-layers and do not use that (require ...) statement. It will work after you `SPC P o e` for now. Sorry for this I will look deeper at this layer.

**Comment by BrianHicks** [comment]
That doesn't seem to be working for me. After opening spacemacs' `init.el` with `PSC P o e`, opening a new project with `SPC p s` just opens the new files in the `main` perspective.

[my current .spacemacs for help in debugging](https://gist.github.com/BrianHicks/f6d8227c56ceb9738e6e)

**Comment by CestDiego** [comment]
Oh you are right....weird it was working like this yesterday D:

**Comment by syl20bnr** [comment]
Fixed with b2893c843ecd84808726375d38ea52feee7fbfa2
Thank you @CestDiego

**Comment by x-ji** [comment]
It doesn't seem to be working again?... I set 
```elisp
     (perspectives :variables
                   perspective-enable-persp-projectile t)
```
according to the documentation in `contrib/perspectives` but seems that no new perspective is being created when I switch projects using `SPC p p`, although I ran `SPC L o e` first.

**Comment by andyque** [comment]
@syl20bnr  I can confirm the issue @x-ji  opened above.
Hope it will be fixed soon.
thanks.

**Comment by x-ji** [comment]
Not only it doesn't work, it seems to also mess up `projectile` cache or something... Now whenever I relaunch Emacs and press `SPC p p`, the project list will always be empty. I just removed `perspectives` layer to get it working again.

**Comment by CestDiego** [comment]
@x-ji I don't have this behaviour in develop. But I agree that some of this functionality is very broken. the fact that perspectives does't lazy-load very well is one of them. I will look at this today.

**Comment by tuhdo** [comment]
Maybe you want to look at the up coming `eyebrowse` package. It works like i3 window manager and for that reason, you can expect the behavior of Vim tabs. To create a new workspace, it's as easy as `SPC W <number>`. The current default behavior is cloning current window configuration, but after this PR is merged: https://github.com/syl20bnr/spacemacs/pull/1384, you will get a clean state when you create a new workspace.

**Comment by CestDiego** [comment]
that's cool, ;D Havin the hability to execute any function indeed seems cool is there a thing like eyebrose-projectile?

**Comment by tuhdo** [comment]
Both could co-exist. `eyebrowse` stores its own window configuration and global buffers; `perspective` stores its own window configuration and **local buffers**. So, you can have generic workspace with all buffers and project workspace with just a set of buffers. However, `eyebrowse` is gonna be the default in Spacemacs to replace Vim tabs, since it implements the equivalent of Vim tabs. You should give `eyebrowse` microstate a try.

**Comment by CestDiego** [comment]
awesum..will do :) thx
